### PR TITLE
apple: ui refinements

### LINKED ISSUE #30: Markdown Editing in iOS
Implement some sort of markdown editor and view

### LINKED ISSUE #398: iOS UI tweaks
Goals:
+ More in line with system applications like Notes, Mail & Calendar
+ More communicative of current state

### PR DESCRIPTION
**Pull request by Parth**
# Bottom bar tweak -- 05a099f

<details>

Before:
image

After:
image

Motivation:

The default place people look for the "new" action on iPhone is bottom right.

Drive:
image
Twitter:
image
Notes:
image
</details>

# Don't recycle view, don't use a List -- 854c2e4

<details>
Eliminates the following weird animation artifact:

animation-artifact

Eliminates this random hole in our list of files:

image

+ Also allows us to get the swipe backwards gesture for free.
+ Also eliminates strange touch zone caused by `Button` / `NavigationLink` discrepancy. 

After:
image

</details>

# Re-arrange icons -- f88d771

<details>
Oriented the icons so 
+ nothing is in the top left (hard to reach)
+ all are the same fill style
+ I don't think we're going to have an "account" page, we're going to think about it as settings (everywhere)
+ this also means the icons don't move around during navigation

imagee
</details>

# Communicating state through the bottom bar -- f88d771
<details>
Serves two purposes:

+ Disable certain behavior during a sync (don't sync during a sync)
+ Communicate state 

## Normal resting state when you've synced:

image

## Offline

image

## Before a sync

image

## During a sync

image
</details>

# Initial macOS app and markdown rendering on all iOS/iPadOS/macOS:

<details>

image

</details>

# Other stability improvements
<details>

Handle when a file is deleted underneath you.

image
</details>

Closes #398 
Closes #30

### CODE REVIEW

File: clients/apple/lockbook.xcodeproj/xcshareddata/xcschemes/Lockbook (iOS).xcscheme [hunk 1]
```diff
@@ -31,9 +31,9 @@
       </Testables>
    </TestAction>
    <LaunchAction
-      buildConfiguration = "Development"
-      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
-      selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
+      buildConfiguration = "Debug"
+      selectedDebuggerIdentifier = ""
+      selectedLauncherIdentifier = "Xcode.IDEFoundation.Launcher.PosixSpawn"
```
Comment (raayan) [inline clients/apple/lockbook.xcodeproj/xcshareddata/xcschemes/Lockbook (iOS).xcscheme:36]: this

File: clients/apple/Shared/Components/BottomBar.swift [hunk 2]
```diff
@@ -0,0 +1,199 @@
+//
+//  BottomBar.swift
+//  Lockbook
+//
+//  Created by Parth Mehrotra on 1/9/21.
+//
```
Comment (raayan) [inline clients/apple/Shared/Components/BottomBar.swift:6]: Remove headers

File: clients/apple/Shared/Views/Book/FileListView.swift [hunk 3]
```diff
@@ -122,58 +62,65 @@ struct FileListView: View {
...
+                    FileCell(meta: meta)
+                    
+                }.isDetailLink(false)
+                .contextMenu(menuItems: {
+                    Button(action: {
+                        handleDelete(meta: meta)
+                    }) {
+                        Label("Delete", systemImage: "trash.fill")
+                    }
+                })
```
Comment (raayan) [inline clients/apple/Shared/Views/Book/FileListView.swift:79]: Put context menu on the returned `AnyView` not each item unless we expect to have diverging menus

**Review by raayan — CHANGES_REQUESTED** [review]
very good

File: clients/apple/Shared/Components/BottomBar.swift (hunk 2 above)
Comment (Parth) [reply clients/apple/Shared/Components/BottomBar.swift]: 04d44f49

File: clients/apple/Shared/Views/Book/FileListView.swift (hunk 3 above)
Comment (Parth) [reply clients/apple/Shared/Views/Book/FileListView.swift]: fd6ff2c9

File: clients/apple/Shared/Components/BottomBar.swift [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+                        Image(systemName: "arrow.triangle.2.circlepath.circle.fill")
+                        #else
+                        Text("Sync now")
+                            .font(.callout)
+                            .foregroundColor(Color.init(red: 0.3, green: 0.45, blue: 0.79))
+                        #endif
+                    })
+                }
+            }
+        }
```
Comment (raayan) [inline clients/apple/Shared/Components/BottomBar.swift:72]: Can this not be encapsulated to a sub view? Xcode and myself like small chunks

File: clients/apple/Shared/Components/BottomBar.swift [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+                    Label("Add", systemImage: "plus.circle.fill")
+                        .imageScale(.large)
+                        .frame(width: 40, height: 40)
+                }
+                )
+            }
+        }
+        #endif
+        
+        // Consider syncing onAppear here
```
Comment (raayan) [inline clients/apple/Shared/Components/BottomBar.swift:47]: remove this

File: clients/apple/Shared/Components/BottomBar.swift [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+            Spacer()
+            statusText
+            Spacer()
+            menu
+            #else
+            Divider()
+            statusText
+            syncButton
+                .padding(.bottom, 4)
+            #endif
```
Comment (raayan) [inline clients/apple/Shared/Components/BottomBar.swift:136]: This is good, what an `#if` should look like

File: clients/apple/Shared/Components/BottomBar.swift [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+                        }
+                    case .Unexpected(_):
+                        core.handleError(err)
+                    }
+                }
+            }
+        }
+    }
+    
+    #if os(iOS)
```
Comment (raayan) [inline clients/apple/Shared/Components/BottomBar.swift:159]: Why in an `#if` ?

File: clients/apple/Shared/Components/FileCell.swift [hunk 8]
```diff
@@ -3,14 +3,30 @@ import SwiftLockbookCore
...
+            .padding(.vertical, 5)
+            Spacer()
+            
+            if meta.fileType == .Folder {
+                Image(systemName: "chevron.right")
+                    .padding(.trailing, 10)
+                    .foregroundColor(.secondary)
+            }
         }
+        .contentShape(Rectangle()) // https://stackoverflow.com/questions/57258371/swiftui-increase-tap-drag-area-for-user-interaction
```
Comment (raayan) [inline clients/apple/Shared/Components/FileCell.swift:29]: `///` for documentation

File: clients/apple/Shared/Config/LockbookMarkdownConfig.swift [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+let emphasisTraits: UIFontDescriptor.SymbolicTraits = [.traitItalic]
+let boldEmphasisTraits: UIFontDescriptor.SymbolicTraits = [.traitBold, .traitItalic]
+let secondaryBackground = UIColor.secondarySystemBackground
+let lighterColor = UIColor.lightGray
+let textColor = UIColor.label
+let headingColor = UIColor(red: 0.94, green: 0.51, blue: 0.69, alpha: 1)
+#endif
+
+// HighlightedTextEditor.markdown will only be accessible via [HighlightRule].markdown in a future 2.0.0 breaking release
+// It does not really make any sense to have it here
```
Comment (raayan) [inline clients/apple/Shared/Config/LockbookMarkdownConfig.swift:48]: `///`

File: clients/apple/Shared/Views/Book/BookView.swift [hunk 10]
```diff
@@ -2,73 +2,27 @@ import SwiftUI
...
-            BookView(core: Core(), account: .fake(username: "jeff"))
-                .ignoresSafeArea()
-//            BookView(core: Core(), account: .fake(username: "test"))
-//                .ignoresSafeArea()
-//                .preferredColorScheme(.dark)
+//                        BookView(core: Core(), account: .fake(username: "jeff"))
+//                            .ignoresSafeArea()
+//                        BookView(core: Core(), account: .fake(username: "test"))
+//                            .ignoresSafeArea()
+//                            .preferredColorScheme(.dark)
```
Comment (raayan) [inline clients/apple/Shared/Views/Book/BookView.swift:25]: Why?

File: clients/apple/macOS/FileListView.swift [hunk 11]
```diff
@@ -0,0 +1,36 @@
+//
+//  FileListView.swift
+//  Lockbook (macOS)
+//
+//  Created by Parth Mehrotra on 1/11/21.
+//
+
```
Comment (raayan) [inline clients/apple/macOS/FileListView.swift:7]: Remove headers plz

File: clients/apple/macOS/OutlineView.swift [hunk 12]
```diff
@@ -0,0 +1,167 @@
+//
+//  OutlineView.swift
+//  OutlineView
+//
+//  Created by Toph Allen on 4/13/20.
+//  Copyright © 2020 Toph Allen. All rights reserved.
+//
```
Comment (raayan) [inline clients/apple/macOS/OutlineView.swift:7]: Remove this and link to source or something

**Review by raayan — CHANGES_REQUESTED** [review]
Mainly some 💅  stuff. Good work brother

**Review by raayan — APPROVED** [review]

**Pull request merged**